litestar/handlers/base.py

```diff
@@ -287,14 +339,31 @@ def resolve_dependencies(self) -> dict[str, Provide]:
             self._resolved_dependencies = {}
 
             for layer in self.ownership_layers:
-                for key, value in (layer.dependencies or {}).items():
-                    if not isinstance(value, Provide):
-                        value = Provide(value)
+                for key, provider in (layer.dependencies or {}).items():
+                    if not isinstance(provider, Provide):
+                        provider = Provide(provider)
+
                     self._validate_dependency_is_unique(
-                        dependencies=self._resolved_dependencies, key=key, provider=value
+                        dependencies=self._resolved_dependencies, key=key, provider=provider
                     )
-                    self._resolved_dependencies[key] = value
 
+                    if not getattr(provider, "signature_model", None):
+                        provider.signature_model = SignatureModel.create(
+                            dependency_name_set=self.dependency_name_set,
+                            fn=provider.dependency.value,
+                            parsed_signature=ParsedSignature.from_fn(
+                                unwrap_partial(provider.dependency.value), self.resolve_signature_namespace()
+                            ),
+                            data_dto=self.resolve_data_dto(),
+                            type_decoders=self.resolve_type_decoders(),
+                        )
+                    if provider.sync_to_thread and not is_async_callable(provider.dependency.value):
+                        provider.dependency.value = async_partial(provider.dependency.value)
+                        provider.has_sync_callable = False
+                    else:
+                        provider.has_sync_callable = not is_async_callable(provider.dependency.value)
+
+                    self._resolved_dependencies[key] = provider
```

Comment:
This should not be a side effect of resolving layered dependencies.

Reply:
Why?

Reply:
Side effects like these from properties are unexpected and obscure what's actually going on. I'm not sure what the motivation was to pull these into the resolve methods instead of having explicit methods for this stuff?

It strikes me as more of a personal style preference.

Reply:
DTOs are instantiated using a class method after resolution, and are cached. 

You are right that it can be abstracted into a sub method, but it still needs to be called at this point in the control flow and exactly once. 

I prefer having the logic in one method rather than spreading it into two, with one calling the other.